pyemma/coordinates/util/patches.py

```diff
@@ -120,3 +160,27 @@ def iterload(filename, chunk=100, **kwargs):
                     raise StopIteration()
 
                 yield traj
+
+
+def _get_local_traj_object(atom_indices, extension, f, n_frames, topology, **kwargs):
+    if extension not in _TOPOLOGY_EXTS:
+        traj = f.read_as_traj(topology, n_frames=n_frames, stride=1, atom_indices=atom_indices, **kwargs)
+    else:
+        traj = f.read_as_traj(n_frames=n_frames, stride=1, atom_indices=atom_indices, **kwargs)
+    return traj
+
+
+def _efficient_traj_join(trajs):
+    assert trajs
+    top = trajs[0].top
+    n_frames = sum(t.n_frames for t in trajs)
+
+    xyz = np.empty((n_frames, top.n_atoms, 3))
+    t = 0
+    for traj in trajs:
+        n = len(traj)
+        xyz[t:n+t] = traj.xyz
+        t += n
+        # TODO: what about time?
```

Comment:
I think it would be useful to append other arguments (since they will be available), as done in reader_utils.py of  https://github.com/gph82/PyEMMA/commit/c179aa7b364d0d71750ecc6a51abe552ecd322e3/ costs nearly nothing and makes the frame complete